File: core-examples/src/main/rb/io/vertx/example/core/verticle/worker/main_verticle.rb
puts "[Main] Running in #{Java::JavaLang::Thread.current_thread().get_name()}"
$vertx.deploy_verticle("io.vertx.example.core.verticle.worker.WorkerVerticle", {
  'worker' => true
})

$vertx.event_bus().send("sample.data", "hello vert.x") { |r_err,r|
  puts "[Main] Receiving reply ' #{r.body().to_string()}' in #{Java::JavaLang::Thread.current_thread().get_name()}"
}

File: core-examples/src/main/js/io/vertx/example/core/verticle/worker/main_verticle.js
console.log("[Main] Running in " + Java.type("java.lang.Thread").currentThread().getName());
vertx.deployVerticle("io.vertx.example.core.verticle.worker.WorkerVerticle", {
  "worker" : true
});

vertx.eventBus().send("sample.data", "hello vert.x", function (r, r_err) {
  console.log("[Main] Receiving reply ' " + r.body() + "' in " + Java.type("java.lang.Thread").currentThread().getName());
});
